### ISSUE TITLE
JS appending empty tfoot elements to tables

### ISSUE THREAD

**Issue by Trott** [TablePress/TablePress#211]
In TablePress 3.0.1, jquery.datatables.min.js adds a `tfoot` element to a table even if the footer content is empty.

An empty `<tfoot>` is an accessibility issue (specifically, WCAG A 1.3.1: "Info and Relationships", in particular the link under "Test Rules" to [ARIA required owned elements](https://www.w3.org/WAI/standards-guidelines/act/rules/bc4a75/proposed/)).

I don't think the unminimized source code is available in this repository (is it?), but the code appending the `tfoot` element is ` (n = H('<tfoot/>').appendTo(r)),`.

Relevant table option settings screenshot:

![image](https://github.com/user-attachments/assets/03e61654-f9e7-4ded-803f-14f73fd8cd4c)

Screenshot of DOM inspector for the table after the JS appends the `<tfoot>`:

![image](https://github.com/user-attachments/assets/5dfaaebc-52cd-4df2-b9d8-878911b59430)

**Comment by TobiasBg** [comment]
Hi @Trott,

Thanks for opening this issue, I really appreciate it

As the mentioned DataTables JavaScript library is an external library (see https://datatables.net/ and https://github.com/DataTables/DataTablesSrc), it's possible that there is nothing really that I can do about this in the end (meaning that this might need to be reported directly to DataTables), but I'll be happy to take a look, of course!

Now, it seems that I can't reproduce this... When I, for example, inspect the small demo table directly on the tablepress.org home page, this does not have a `<tfoot>` element, neither in Google Chrome nor in Firefox on a Mac, for me. Can you maybe verify that?

Best wishes,
Tobias

**Comment by TobiasBg** [comment]
Hi @Trott,

quick update: Please scratch what I wrote above. I was able to reproduce this now. The mentioned table has "Horizontal Scrolling" enabled, which results in a slightly different DOM structure.

Unfortunately, I don't really see what I could do about this directly in TablePress. I'm not really fond of patching the used DataTables JS file, so ideally this would have to be fixed directly in DataTables...

Best wishes,
Tobias

**Comment by Trott** [comment]
OK, thanks! I'll close this.

**Comment by TobiasBg** [comment]
Hi @Trott,

do you maybe want to open this issue at https://github.com/DataTables/DataTablesSrc
I'm sure, Allan, the DataTables developer will be happy to take a look!

Best wishes,
Tobias

**Comment by Trott** [comment]
@AllanJard Is this an issue I should open in the DataTables repo? Or would you prefer I open something in the support forum instead? It seems to me that https://github.com/DataTables/DataTablesSrc/blob/49e7219c8a32ba04d7954494b392276ef18115f3/js/core/core.constructor.js#L338-L340 should only run if there will be a caption. But maybe it's not possible for DataTables to know if there will be a caption or not? Happy to open an issue if the problem is truly in DataTables (which I suspect, but my knowledge is superficial) and not that I'm doing something wrong.

**Comment by AllanJard** [comment]
Yes, that looks like an error in DataTables. Happy for you to open it in the github repo or in the forum.

I think I probably added it thinking it would be harmless and it was easier to just add (particularly as column footers can be added dynamically via the init options), but apparently it isn't.

**Comment by Trott** [comment]
OK, I opened https://github.com/DataTables/DataTablesSrc/issues/308 and I imagine any additional conversation should be over there. Thanks!
